src/scenes/ObservationsRouter/details/index.tsx

```diff
@@ -196,8 +205,25 @@ export default function ObservationDetails(props: ObservationDetailsProps): JSX.
       .map((sp) => {
         return { otherSpeciesName: sp.otherSpeciesName!, speciesId: sp.speciesId! };
       });
-    const request = dispatch(requestMergeOtherSpecies({ mergeOtherSpeciesPayloads: merged, observationId }));
-    setMergeRequestId(request.requestId);
+
+    if (merged.length > 0) {
+      const successMessageBody = (
+        <ul style={{ paddingLeft: '24px', margin: 0 }}>
+          {merged.map((sp, index) => {
+            const newName = allSpecies?.find((existing) => existing.id === sp.speciesId)?.scientificName;
+            return (
+              <li key={index}>
+                {sp.otherSpeciesName} &#8594; {newName}
+              </li>
+            );
+          })}
+        </ul>
+      );
+      setMergedMessage(successMessageBody);
```

Comment:
Storing components in state is generally not advised. I think you should create and load the component into the snackbar at the time of the toast.

Reply:
Something like this could solve it: https://github.com/terraware/terraware-web/pull/3350.

But I am unsure if that code actually works with what you're attempting. What you propose, saving the names into state, would probably also work just as well. 

I also think that since this is the only place `requestMergeOtherSpecies`, a cleaner approach would be to handle the "payload creation logic" in the thunk. This way the component really just needs to dispatch the request, and the result could contain the filtered payloads with the names. 

Ideally a component doesn't need to manage to pieces of separate state for a request if the request itself can contain all the data necessary to satisfy the UI.

Reply:
Here's an example of that: https://github.com/terraware/terraware-web/pull/3351/files

Again, this is not tested, so it may not satisfy your needs. Anyways, these are just suggestions. Let me know what you think. I don't want to make you completely redo this effort, so if what I'm asking for is off base just let me know!